File: app/src/main/java/me/tagavari/airmessage/util/DisposableViewHolder.java
package me.tagavari.airmessage.util;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import io.reactivex.rxjava3.disposables.CompositeDisposable;

/**
 * A view holder that keeps track of a {@link CompositeDisposable} for background processes
 */
public abstract class DisposableViewHolder extends RecyclerView.ViewHolder {
	private final CompositeDisposable compositeDisposable = new CompositeDisposable();
	
	public DisposableViewHolder(@NonNull View itemView) {
		super(itemView);
	}
	
	public CompositeDisposable getCompositeDisposable() {
		return compositeDisposable;
	}
}

File: app/src/main/java/me/tagavari/airmessage/util/DisposableViewHolder.kt
package me.tagavari.airmessage.util

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import io.reactivex.rxjava3.disposables.CompositeDisposable

/**
 * A view holder that keeps track of a [CompositeDisposable] for background processes
 */
abstract class DisposableViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
	val compositeDisposable = CompositeDisposable()
}